### PR TITLE
Add DIRHTML builder support

### PR DESCRIPTION
**Pull request by sitic**
Adds support the Sphinx `dirhtml` builder, in addition to the `html` builder. 

The `dirhtml` builder builds HTML pages, but with a single directory per document. Makes for prettier URLs (no .html) if served from a webserver.

- [x] Tests written and passed
- [x] Documentation and changelog entry written, docs build passed
- [x] All `tox` checks passed

### CODE REVIEW

File: tests/extension/__init__.py [hunk 1]
```diff
@@ -46,7 +47,8 @@ def assert_links(file: Path, links: list):
 
 
 @pytest.mark.parametrize(("name", "file"), ref_tests)
-def test_references(name: str, file: Path, tmp_path: Path):
+@pytest.mark.parametrize("builder", builders)
```
[the remark this answers is not in the document]
Comment (sitic) [reply tests/extension/__init__.py:50]: I agree, I wanted to be sure I haven't missed an edge case.

The important check for coverage is a call to `check_link_targets()` to verify that the correct links have been set. Currently, this check is only being performed in the `test_references()` and `test_parallel_build()` tests.

Would you prefer either:
1. Adding a separate dedicated test case for the `dirhtml` builder, or
2. Keeping the `@pytest.mark.parametrize("builder", builders)` only for `test_parallel_build()` and removing it from other tests?

File: tests/extension/__init__.py [hunk 2]
```diff
@@ -243,13 +250,14 @@ def raise_nomsg(*_, **__):
...
+        _sphinx_build(tmp_path, builder, files)
 
     with pytest.raises(RuntimeError), patch(target, raise_nomsg):
-        _sphinx_build(tmp_path, "html", files)
+        _sphinx_build(tmp_path, builder, files)
 
 
-def test_parallel_build(tmp_path: Path):
+@pytest.mark.parametrize("builder", builders)
+def test_parallel_build(builder: str, tmp_path: Path):
```
Comment (sitic) [inline tests/extension/__init__.py:260]: Here is the `test_parallel_build()` case

File: tests/extension/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sitic) [reply tests/extension/__init__.py:50]: I agree, I wanted to be sure I haven't missed an edge case.

The critical part of our verification is the `check_link_targets()` call, which confirms that the correct links have been established.  Currently, this check is only being performed in the `test_references()` and `test_parallel_build()` tests.

For better coverage, would you prefer either:
1. Adding a separate dedicated test case for the `dirhtml` builder, or
2. Keeping the `@pytest.mark.parametrize("builder", builders)` only for `test_parallel_build()` and removing it from other tests?

[the remark this answers is not in the document]
Comment (felix-hilden) [reply tests/extension/__init__.py:50]: I think if the only difference between the builders is where they place the file (i.e. file/index.html vs file.html) then we don't need to duplicate all of it. We could just write a new pytest routine here to test dirhtml, and use a comprehensive enough set of test sources (I'd assume at least one non-index doc). But is there something else we should test?

**Comment by felix-hilden** [comment]
Just pushed a small formatting change, thank you for the contribution and I hope the extension is useful to you!

**Comment by felix-hilden** [comment]
0.17.3 also released now!
